reconcile/utils/terraform_config.py

```diff
@@ -0,0 +1,104 @@
+from typing import Iterable, Optional, Protocol
+from reconcile.utils.external_resource_spec import (
+    ExternalResourceSpecInventory,
+    ExternalResourceSpec,
+)
+
+from reconcile.utils.ocm import OCMMap
+
+
+class UnknownProvisionProviderError(Exception):
+    pass
+
+
+class ProvisionProviderAlreadyRegisteredError(Exception):
+    pass
+
+
+class TerraformClientProtocol(Protocol):
+    def dump(
+        self,
+        print_to_file: Optional[str] = None,
+        existing_dirs: Optional[dict[str, str]] = None,
+    ) -> dict[str, str]:
+        ...
+
+    def init_populate_specs(
+        self, specs: Iterable[ExternalResourceSpec], account_name: Optional[str]
+    ) -> None:
+        ...
+
+    def populate_resources(
+        self, spec: ExternalResourceSpec, ocm_map: Optional[OCMMap] = None
+    ):
+        ...
+
+
+class TerraformConfigProvider:
+    """
+    At a high-level, this class is responsible for generating Terraform configuration in
+    JSON format from app-interface schemas/openshift/external-resource-1.yml objects.
```

Comment:
In theory that is the idea as long as the Terraform CDK, or whatever other technology, could implement the methods in `TerraformClientProtocol`. I'll probably move this to an abstract class because we should have direct control over all of these.

Reply:
In this model, is it correct to say there will be `TerrascriptAwsClient`, `TerraformCDKAwsClient` etc? In that case, ignore my other comment on composition and adding another layer.